src/matter/session/secure/SecureChannelMessenger.ts

```diff
@@ -60,13 +83,23 @@ export class SecureChannelMessenger<ContextT> {
         await this.exchange.send(MessageType.StatusReport, writer.toByteArray());
     }
 
-    protected throwIfError(messageType: number, payload: ByteArray) {
-        if (messageType !== MessageType.StatusReport) return;
+    decodeStatusReport(payload: ByteArray) {
         const reader = new DataReader(payload, Endian.Little);
         const generalStatus = reader.readUInt16();
-        if (generalStatus === GeneralStatusCode.Success) return;
         const protocolId = reader.readUInt32();
         const protocolStatus = reader.readUInt16();
+        return { generalStatus, protocolId, protocolStatus };
+    }
+
+    protected throwIfError(messageType: number, payload: ByteArray) {
+        if (messageType !== MessageType.StatusReport) return;
+        const { generalStatus, protocolId, protocolStatus } = this.decodeStatusReport(payload);
+        if (generalStatus === GeneralStatusCode.Success) {
```

Comment:
Avoid unnecessary nesting:
if (generalStatus !== GeneralStatusCode.Success) throw new Error(`Received error status: ${generalStatus} ${protocolId} ${protocolStatus}`);
if (protocolStatus !== ProtocolStatusCode.Success) throw new Error(`Received success status, but protocol status is not Success: ${protocolStatus}`);